hops/dataset.py

```diff
@@ -214,3 +217,195 @@ def download(remote_path, file, chunk_size=None):
+        :project_name: whether this method should wait for the zipping process to complete beefore returning.
+
+    Returns:
+        True if path exists, False otherwise
+    """
+    try:
+        get_path_info(remote_path, project_name)
+        return True
+    except RestAPIError:
+        return False
+
+
+def delete(remote_path, project_name=None, block=True, timeout=30):
+    """
+    Delete the dir or file in Hopsworks, specified by the remote_path.
+
+    Example usage:
+
+    >>> from hops import dataset
+    >>> dataset.delete("Projects/project_name/Resources/myremotefile.txt")
+
+    Args:
+        :remote_path: the path to the remote file or directory in the dataset
+        :project_name: whether this method should wait for the zipping process to complete beefore returning.
+        :block: whether to wait for the deletion to complete or not.
+        :timeout: Number of seconds to wait for the deletion to complete before returning.
+
+    Returns:
+        None
+    """
+    project_id = project.project_id_as_shared(project_name)
+    resource_url = constants.DELIMITERS.SLASH_DELIMITER + \
+                   constants.REST_CONFIG.HOPSWORKS_REST_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER + \
+                   constants.REST_CONFIG.HOPSWORKS_PROJECT_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER + \
+                   project_id + constants.DELIMITERS.SLASH_DELIMITER + \
+                   constants.REST_CONFIG.HOPSWORKS_DATASETS_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER + \
+                   remote_path
+
+    util.send_request('DELETE', resource_url)
+
+    # Check if path is indeed deleted as REST API is asynchronous
+    if block:
+        count = 0
+        while count < timeout and path_exists(remote_path, project_name):
+            print("Waiting for deletion...")
+            count += 1
+            time.sleep(1)
+
+        if count >= timeout:
+            raise DeletionTimeout("Timeout of {} seconds exceeded while deleting path {}.".format(timeout, remote_path))
+
+
+def compress(remote_path, project_name=None, block=False, timeout=120):
+    """
+    Compress the dir or file in Hopsworks, specified by the remote_path.
+
+    Example usage:
+
+    >>> from hops import dataset
+    >>> dataset.compress("Projects/project_name/Resources/myremotefile.txt")
+
+    Args:
+        :remote_path: the path to the remote file or directory in the dataset
+        :project_name: whether this method should wait for the zipping process to complete beefore returning.
+        :block: whether to wait for the compression to complete or not.
+        :timeout: Number of seconds to wait for the compression to complete before returning.
+
+    Returns:
+        None
+    """
+    _archive(remote_path, action='zip', block=block, project_name=project_name, timeout=timeout)
+
+
+def extract(remote_path,  block=False, project_name=None, timeout=120):
```

Comment:
Although they are optional, I'd suggest being consistent with the order of the parameters in `compress`, `extract` and `archive` methods. We can use the same order as in `compress`: `project_name, block and timeout`.

Reply:
Fixed